### DISCUSSION TITLE
Using voxels

### DISCUSSION THREAD

**Discussion by whtemple1959** [TokisanGames/Terrain3D#337] [Questions] [answered]
I am just a hobbyists for now just beginning to study C++ and have chosen Godot as the game engine of choice .
I am interested in developing a top down strategy game with a terrain some where between Minetest and real world. Much like what is created with the plugin retro terrain.
But, I do not want to try and learn C# while trying to learn C++
I am using voxel tool version of Godot but I am still struggling with the heigthmap integration. So my question is. Can the terrain be manipulated in Terrain 3D to appear blocky?
![what_game_terrain_needs_to_look_like](https://github.com/TokisanGames/Terrain3D/assets/70185317/a3568fb8-f350-43cf-ad2d-0f18ab0e0654)

**Comment by TokisanGames** [comment, marked as answer]
To match your image, you can customize the shader to add your own vertex grid as demonstrated in the vertex grid debug shader, and use more uniform, non-realistic textures. 

![image](https://github.com/TokisanGames/Terrain3D/assets/632766/ca9e45fb-d7c0-4cb6-a8e1-1ad38e8f1598)


For a low poly look, you could change texture filtering to nearest and reduce the UV scale of the textures or use low res textures if you want blocky textures (both shown). You could modify your custom shader to use flat normals, or adjust the texture blending to limit it to one texture per square (neither shown here, nor in your image).

![image](https://github.com/TokisanGames/Terrain3D/assets/632766/5c131985-35af-400c-8af0-302d53434f7e)

> Can the terrain be manipulated in Terrain 3D to appear blocky?

Making a blocky terrain like minecraft would require adding vertices, which cannot be done by the shader. It would require rewriting our mesh generation. Instead, you should use voxel tools, which is designed for it.

> I am using voxel tool version of Godot but I am still struggling with the heigthmap integration

You should ask about that on Zylann's repository and learn to use his tool better. It works fine with heightmaps.

**Reply by TokisanGames** [reply]
Closing due to question answered and conversation stopped.
